templates/terraform/iam_policy.go.erb

```diff
@@ -119,8 +119,6 @@ func <%= resource_name -%>IamUpdaterProducer(d *schema.ResourceData, config *Con
 <% end -%>
 <% end -%>
 
-	d.SetId(u.GetResourceId())
```

Comment:
I'm assuming this was redundant with the caller?

Reply:
Yup, and since the IDs can contain more information in them now than just the resourceId (like the role for example), it was setting it to the wrong value.